### DISCUSSION TITLE
browsershot & laravel-pdf

### DISCUSSION THREAD

**Discussion by azc666** [spatie/laravel-pdf#68] [Q&A] [answered]
I am using laravel 10 jetstream php 8.2. Installed puppeteer, chrome, and then spatie/laravel-pdf. Browsershot works but not laravel-pdf. Any ideas for me? btw ->newHeadless() makes browsershot work, but is not a method in Pdf.

Route::get('/testes', function () {
    $name = 'Peter Parker';
    \Spatie\LaravelPdf\Facades\Pdf::view('pdf.testes')
    ->paperSize(3.5, 2, 'in')
    // ->newHeadless()
    ->save('bc-cardtest5.pdf');

    //  this works as well //
    // $pdf = new Spatie\PdfToImage\Pdf('bc-cardtest4.pdf');
    // $pdf->saveImage('bc-cardtest4.jpg');

    Browsershot::url('https://example.com')
        ->newHeadless()
        ->save('example3.pdf');
});

**Comment by ArielMejiaDev** [comment, marked as answer]
@azc666 you can add Browsershot instance to add more customization using Laravel PDF, here is an example from the [docs](https://spatie.be/docs/laravel-pdf/v1/advanced-usage/customizing-browsershot), so then you can add `newHeadless()` method and chain more of them using the browsershot instance from the closure, eg:

```php
use Spatie\LaravelPdf\Facades\Pdf;

Pdf::url('https://example.com')
    ->withBrowsershot(function (Browsershot $browsershot) {
        $browsershot->newHeadless();
    })
    ->save('example3.pdf');
```

**Reply by azc666** [reply]
TY !!!!
